core/domain/question_domain.py

```diff
@@ -459,6 +462,183 @@ def _convert_state_v34_dict_to_v35_dict(cls, question_state_dict):
 
         return question_state_dict
 
+    @classmethod
+    def _convert_state_v35_dict_to_v36_dict(cls, question_state_dict):
+        """Converts from version 35 to 36. Version 35 adds translation support
+        for interaction customization arguments. This migration converts
+        customization arguments whose schemas have been changed from unicode to
+        SubtitledUnicode or html to SubtitledHtml. It also populates missing
+        customization argument keys on all interactions, removes extra
+        customization arguments, normalizes customization arguments against
+        its schema, and changes PencilCodeEditor's customization argument
+        name from initial_code to initialCode.
+        """
+        max_existing_content_id_index = -1
+        translations_mapping = question_state_dict[
+            'written_translations']['translations_mapping']
+        for content_id in translations_mapping:
+            # Find maximum existing content_id index.
+            content_id_suffix = content_id.split('_')[-1]
+
+            # Possible values of content_id_suffix are a digit, or from
+            # a 'outcome' (from 'default_outcome'). If the content_id_suffix
+            # is not a digit, we disregard it here.
+            if content_id_suffix.isdigit():
+                max_existing_content_id_index = max(
+                    max_existing_content_id_index,
+                    int(content_id_suffix)
+                )
+
+            # Move 'html' field to 'translation' field and set 'data_format'
+            # to 'html' for all WrittenTranslations.
+            for lang_code in translations_mapping[content_id]:
+                translations_mapping[
+                    content_id][lang_code]['data_format'] = 'html'
+                translations_mapping[
+                    content_id][lang_code]['translation'] = (
+                        translations_mapping[content_id][lang_code]['html'])
+                del translations_mapping[content_id][lang_code]['html']
+
+        interaction_id = question_state_dict['interaction']['id']
+        if interaction_id is None:
+            question_state_dict['next_content_id_index'] = (
+                max_existing_content_id_index + 1)
+            return question_state_dict
+
+        class ContentIdCounter(python_utils.OBJECT):
```

Comment:
Shouldn't this be a part of state_domain, or is it used only when migrating?

Reply:
Only used when migrating